File: leetcode/11.盛最多水的容器.go
func maxArea(height []int) int {
    //注意为什么能够使用双指针解法
    n := len(height)
    l, r := 0, n-1
    ans := 0
    for l < r {
        ans = max(ans, (r - l) * min(height[r], height[l]))
        if height[l] < height[r] {
            l += 1
        }else{
            r -= 1
        }
    }
    return ans
}
func min(a, b int) int{
    if a < b {
        return a
    }
    return b
}
func max(a, b int) int{
    if a < b {
        return b
    }
    return a
}

File: leetcode/11.盛最多水的容器.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n=len(height)
        i,j=0,n-1
        res=-1
        while i<j:
            res=max(res,(j-i)*min(height[i],height[j]))
            if height[i]<=height[j]:
                i+=1
            else:
                j-=1
        return res
